### DISCUSSION TITLE
Add Attributes with dash

### DISCUSSION THREAD

**Discussion by phneoix** [Knio/dominate#174] [Q&A] [answered]
Hi i am using htmx library on frontend. When i try to add htmx attributes to elements (like hx-get='/' or hx-ext) i get error. how do i add attributes with '-'.

dttg.td(dttg.button("DV",_type="button",cls="btn btn-primary btn-sm",hx-ext='ws')

SyntaxError: expression cannot contain assignment, perhaps you meant "=="?

**Comment by Knio** [comment, marked as answer]
You can add attributes like this using a dictionary or the item interface, ie:

```py
b = button(**{'hx-get': '/'})
# or 
b['hx-ext'] = 'ws'
```

**Reply by cj** [reply]
@Knio, first off, thank you for this awesome lib; I just found it!

Regarding @phneoix's question, I was looking at using htmx too. Would another way be to make https://github.com/Knio/dominate/blob/master/dominate/dom_tag.py#L420-L422 special_prefix extendable with some sort of global option? So that way, you could do `button(hx_ext='ws')`?

**Reply by Knio** [reply]
I don't want to get into the habit of supporting special cases for 3rd-party libraries.
However, looks like it would be easy to move the `('data_', 'aria_')` expression into a class attribute that you could just overwrite?
(or without any changes, you could just overwrite the `clean_attribute` function)
